core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/GetWalksHandler.java

```diff
@@ -0,0 +1,147 @@
+        if (null == getWalks.getInput()) {
+            return null;
+        }
+
+        // Check operations input
+        final Iterable<GetElements> operations = getWalks.getOperations();
+
+        if (null == operations || Iterables.isEmpty(operations)) {
+            return new EmptyClosableIterable<>();
+        }
+
+        hops = getWalks.getOperations().size();
+
+        final List<AdjacencyList<Object, Edge>> adjacencyLists = new ArrayList<>();
+
+        List<Edge> results = null;
+
+        // Execute the GetElements operations
+        for (final GetElements getElements : operations) {
+            if (null == results) {
+                getElements.setInput(getWalks.getInput());
+
+                // Cache the results locally
+                results = getLimitedResults(store, context, getWalks, getElements);
+            } else {
+                final OperationChain<CloseableIterable<? extends Element>> opChain = new OperationChain.Builder()
+                        .first(new ToVertices.Builder()
+                                .input(results)
+                                .useMatchedVertex(ToVertices.UseMatchedVertex.OPPOSITE)
+                                .build())
+                        .then(new ToEntitySeeds())
+                        .then(getElements)
+                        .build();
+
+                // Cache the results locally
+                results = getLimitedResults(store, context, getWalks, opChain);
+            }
+
+            final AdjacencyList<Object, Edge> adjacencyList = new AdjacencyList<>();
+
+            // Store results in an AdjacencyList
+            for (final Edge e : results) {
+                adjacencyList.put(e.getMatchedVertexValue(), e.getAdjacentMatchedVertexValue(), e);
+            }
+
+            adjacencyLists.add(adjacencyList);
+        }
+
+        // Track/recombine the edge objects and convert to return type
+        return Streams.toStream(getWalks.getInput())
+                .map(seed -> walk(seed.getVertex(), null, adjacencyLists, new LinkedList<>()))
+                .flatMap(List::stream)
+                .collect(Collectors.toList());
+    }
+
+    private List<Walk> walk(final Object curr, final Object prev, final List<AdjacencyList<Object, Edge>> adjacencyLists, final LinkedList<Set<Edge>> queue) {
+        final List<Walk> walks = new ArrayList<>();
+
+        if (null != prev && hops != queue.size()) {
+            queue.offer(adjacencyLists.get(queue.size()).get(prev, curr));
+        }
+
+        if (hops == queue.size()) {
+            final Walk.Builder builder = new Walk.Builder();
+            for (final Set<Edge> edgeSet : queue) {
+                builder.edges(edgeSet);
+            }
+            final Walk walk = builder.build();
+            walks.add(walk);
+        } else {
+            for (final Object obj : adjacencyLists.get(queue.size()).getDestinations(curr)) {
+                walks.addAll(walk(obj, curr, adjacencyLists, queue));
```

Comment:
I'm concerned that we could get a StackOverflowError here if there are a large number of edges coming off a single vertex.

Reply:
The number of stack frames is correlated with the number of hops, rather than the number of vertices, so this shouldn't be an issue.